### PR TITLE
[Key Vault] Remove direct uses of `asyncio.sleep`

### PR DESCRIPTION
**Pull request by mccoyp**
# Description

Resolves https://github.com/Azure/azure-sdk-for-python/issues/33748. Key Vault uses a handful of custom polling methods, which currently use `asyncio.sleep` directly. This PR refactors them to instead use the client transport's own `sleep` method. Polling methods now accept an operation's original pipeline response as a parameter and use its pipeline context.

`asyncio` is still imported and used directly in tests and samples; a separate PR will follow this one to remove test instances (to keep this PR as simple as possible).

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py [hunk 1]
```diff
@@ -120,11 +121,13 @@ async def create_certificate(
...
         create_certificate_polling = CreateCertificatePollerAsync(
-            get_certificate_command=get_certificate_command, interval=polling_interval
+            pipeline_response=pipeline_response,
+            get_certificate_command=get_certificate_command,
+            interval=polling_interval,
         )
         def no_op(*_, **__) -> Any:  # The deserialization callback is ignored based on polling implementation
             pass
-        return await async_poller(command, create_certificate_operation, no_op, create_certificate_polling)
+        return await AsyncLROPoller(command, create_certificate_operation, no_op, create_certificate_polling)
```
Comment (kristapratico) [inline sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py:130]: I don't think we need an await here where we return the poller object

**Review by kristapratico — APPROVED** [review]
LGTM. Add the fix to the changelog?

File: sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py (hunk 1 above)
Comment (mccoyp) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py:130]: This API's a little odd in that we don't actually return the poller; we return the result of awaiting `.result()` directly (awaiting the poller is just a shorthand): https://github.com/Azure/azure-sdk-for-python/blob/9cb79477740fed1d5057fe6a14cdc0ab10e332d3/sdk/core/azure-core/azure/core/polling/_async_poller.py#L182-L183
This change was just moving away from the deprecated `async_poller` function 🙂

Comment (kristapratico) [reply sdk/keyvault/azure-keyvault-certificates/azure/keyvault/certificates/aio/_client.py:130]: Ah I see! Ignore me!

**Pull request merged**